ocean_lib/assets/asset_downloader.py

```diff
@@ -16,77 +16,62 @@
 logger = logging.getLogger(__name__)
 
 
-# TODO: fix for v4
-
-
 @enforce_types
 def download_asset_files(
-    service_index: int,
     asset: Asset,
+    provider_uri: str,
```

Comment:
the DataProvider must have its own provider_uri, we shouldn't pass it separately.

Reply:
I need provider_uri for building the URL for download and if I don't pass provider_uri as parameter, I need to pass a config object from where I retrieve provider_url as I used in the tests with a fixture config.

Reply:
DataServiceProvider already has a property set. If the property is not set, it will grab the default provider_uri from config. I assure you there is no need for provider_uri if there is a DataServiceProvider object.

Reply:
I will delete DataServiceProvider object and keep just `provider_uri`.